### PR TITLE
[WIP] Facade illumination proposal

### PR DESCRIPTION
**Pull request by clinique**
I'd like your advice regarding this pull-request and the way it is handled, especially I had to move AstroThingHandler to being a bridge in order to attach Facades to a given sun. This by the way currently breaks astro.test package.

Signed-off-by: Gaël L'hopital <gael@lhopital.org>

### CODE REVIEW

**Comment by htreu** [comment]
Hi @clinique thats a pretty neat feature, thanks for bringing this up.
The astro binding however may not be an ideal place for this calculations. The astro binding is hard on the edge of being an acceptable binding itself, since it does not control or communicate with any external device or service. It merely calculates numbers based on a given position and time. The same holds true for the new facade thing here.
As an alternative please try to implement the same calculations in a `TriggerModuleHandler` for the new automation engine. Please refer to the `ItemStateTriggerHandler` and the TriggerModuleType description in `ItemTriggers.json`.

The json file describes the trigger module type with a config description and optional outputs. The handler is the instance which runs the trigger and gets instantiated by the new rule engine.

Your trigger handler would have a configuration where the related `sun` items can be configured. Also, the facade properties (like `orientation`, `offset`s and `margin`) will be part of the config description. On ever state change of one of the items you re-calculate the occurrence of one of the three events (enters, front, leaves) and call `callback.triggered` in your handler.

You may provide all this in a new bundle which then can be extended to add other related stuff, like roof surface area and the like.

WDYT?

**Comment by clinique** [comment]
Hello @htreu , thanks for your feed-back. I nearly not surprized by this. I was aware this would probably not be the best place but wanted to give this a chance to progress. So, I'll try your pointers to see how it can fit.
BR !

**Comment by clinique** [comment]
Is there any 'reference implementation' I can take a look at in order give me some samples to study (0 reference to TriggerModuleHandler in OH2 repo, and 1 in ESH repo :fearful: )

**Comment by htreu** [comment]
The best reference I could find was the `ItemStateTriggerHandler` which corresponds to the `ItemTriggers.json` where the trigger module types are defined. Did you inspect those?

**Comment by clinique** [comment]
Yes, digged into org.eclipse.smarthome.automation.module.xxx package, but what to do is unclear. Sorry.

**Comment by clinique** [comment]
I've got the feeling that we somehow are in the same case than discussed [here](https://github.com/eclipse/smarthome/issues/4748)

**Comment by htreu** [comment]
To be honest, implementing your logic in a automation trigger module will reduce the functionality compared to the current implementation.
The state which is currently available on the "" channel will be gone and only the trigger functionality will be retained. Is this a problem for the use cases you have in mind? Please list some of the use cases, so we can make an educated decision on how to proceed.

**Comment by htreu** [comment]
@clinique this was not meant to thwart you in any way. Maybe we could start by listing some real life examples and use cases, wdyt?
From that we can move on and decide which solution fits best here. I would welcome your participation.

**Comment by clinique** [comment]
@htreu : no problem - don't bother, I have many PR ongoing, this one will wait a bit as it needs much more rethinking.

File: extensions/binding/org.eclipse.smarthome.binding.astro/README.md [hunk 1]
```diff
@@ -86,6 +89,12 @@ Optionally, a refresh `interval` (in seconds) can be defined to also calculate p
         * **channel** 
             * `azimuth, elevation` (Number:Angle)
 
+* **thing** `facade`
+        * **channel**: 
+            * `facingsun` (Switch) : ON if orientation - negative offset <= sun azimut <= orientation + negative offset
+            * `bearing` (Number:Dimensionless) : pourcentage of direct lightning. 100% when the sun is in front, 0% when not facing sun
```
Comment (wborn) [inline extensions/binding/org.eclipse.smarthome.binding.astro/README.md]: pourcentage -> percentage

File: extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/channels.xml [hunk 2]
```diff
@@ -468,5 +468,46 @@
...
+        <item-type>Switch</item-type>
+        <label>Facing Sun</label>
+        <description>Indicates whether the sun is facing facade</description>
+        <state readOnly="true" />
+    </channel-type>
+    
+    <channel-type id="bearing">
+        <item-type>Number:Dimensionless</item-type>
+        <label>Bearing</label>
+        <description>Sun bearing pourcentage (100% when the sun is in facade orientation)</description>
```
Comment (wborn) [inline extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/channels.xml]: pourcentage -> percentage

File: extensions/binding/org.eclipse.smarthome.binding.astro/README.md [hunk 3]
```diff
@@ -20,10 +21,12 @@ No binding configuration required.
 
 ## Thing Configuration
 
-All Things require the parameter `geolocation` (as `<latitude>,<longitude>,[<altitude in m>]`) for which the calculation is done. 
+Sun and Moon Things require the parameter `geolocation` (as `<latitude>,<longitude>,[<altitude in m>]`) for which the calculation is done. 
 The altitude segment is optional and sharpens results provided by the Radiation group.
 Optionally, a refresh `interval` (in seconds) can be defined to also calculate positional data like azimuth and elevation.
 
+Facade requires a bridge reference to the Sun and the orientation of the facade toward geographic north. By default, the facade is considered enlighted when the sun azimuth reaches orientation - 90° (negative offset) until orientation + 90° (positive offset).
+The two parameters (negative and positive offset) are set by default to 90. These can be modified to take in account natural obstacles (wall, trees) that reduces the lightning angle.
```
Comment (wborn) [inline extensions/binding/org.eclipse.smarthome.binding.astro/README.md]: lightning -> lighting

File: extensions/binding/org.eclipse.smarthome.binding.astro/README.md (hunk 1 above)
Comment (wborn) [reply extensions/binding/org.eclipse.smarthome.binding.astro/README.md]: lightning -> lighting

**Comment by clinique** [comment]
I'll take a different approach by using channel profiles.

**Comment by kaikreuzer** [comment]
Then let's close this PR and continue with a new one at https://github.com/openhab/openhab-core.

**Pull request closed without merging**